index.bs

```diff
@@ -485,6 +501,162 @@ spec:csp3; type:grammar; text:base64-value
+      readonly attribute USVString destination;
+      readonly attribute boolean   reportOnly;
+    };
+  </pre>
+
+  To <dfn>report violation</dfn> given a <a for=/>Request</a> |request|, a boolean |block|,
+  a boolean |reportBlock|, an <a>integrity policy struct</a> |policy|,
+  and an <a>integrity policy struct</a> |reportPolicy|, do the following:
+
+  1. <a>Assert</a>: |request|'s <a for=request>client</a> is not null.
+  1. Let |settingsObject| be |request|'s <a for=request>client</a>.
+  1. Let |global| be |settingsObject|'s <a for="environment settings object">global object</a>.
+  1. <a>Assert</a>: |global| is a {{Window}} or a {{WorkerGlobalScope}}.
+  1. Let |url| be null.
+  1. If |global| is a {{Window}}, set |url| to |global|'s <a>associated Document</a>'s {{Document/URL}}.
+  1. If |global| is a {{WorkerGlobalScope}}, set |url| to |global|'s <a for=WorkerGlobalScope>URL</a>.
+  1. <a>Assert</a>: |url| is a <a for=/>URL</a>.
+  1. Let |documentURL| be the result of <a>strip URL for use in reports</a> on |url|.
+  1. Let |blockedURL| be the result of <a>strip URL for use in reports</a> on |request|'s <a for=request>URL</a>.
+  1. If |block|, <a for=list>for each</a> |endpoint| in |policy|'s <a>endpoints</a>:
+     1. Let |body| be a new {{IntegrityPolicyViolationReportBody}}, initialized as follows:
+                 :   {{IntegrityPolicyViolationReportBody/documentURL}}
+                 ::  |documentURL|
+                 :   {{IntegrityPolicyViolationReportBody/blockedURL}}
+                 ::  |blockedURL|
+                 :   {{IntegrityPolicyViolationReportBody/destination}}
+                 ::  |request|'s <a for=request>destination</a>
+                 :   {{IntegrityPolicyViolationReportBody/reportOnly}}
+                 ::  false
+     2. <a>Generate and queue a report</a> with the following arguments:
+              :   <var ignore>context</var>
+              ::  |settingsObject|
+              :   <var ignore>type</var>
+              ::  "`integrity-policy-violation`"
+              :   <var ignore>destination</var>
+              ::  |endpoint|
+              :   <var ignore>data</var>
+              ::  |body|
+  1. If |reportBlock|, <a for=list>for each</a> |endpoint| in |reportPolicy|'s <a>endpoints</a>:
+     1. Let |reportBody| be a new {{IntegrityPolicyViolationReportBody}}, initialized as follows:
+                 :   {{IntegrityPolicyViolationReportBody/documentURL}}
+                 ::  |documentURL|
+                 :   {{IntegrityPolicyViolationReportBody/blockedURL}}
+                 ::  |blockedURL|
+                 :   {{IntegrityPolicyViolationReportBody/destination}}
+                 ::  |request|'s <a for=request>destination</a>
+                 :   {{IntegrityPolicyViolationReportBody/reportOnly}}
+                 ::  true 
+     2. <a>Generate and queue a report</a> with the following arguments:
+              :   <var ignore>context</var>
```

Comment:
I think normally for named arguments we use `<i>` on the call site and we link their definition?

Reply:
I *think* that requires exporting them on the Reporting site in ways that aren't currently there.

Reply:
https://github.com/w3c/reporting/pull/279

Now we have two options:
1) we wait for the reporting API change to land and propagate and then add the correct links here.
2) Land this as is, and then fix it as a followup.

@mozfreddyb - which do you prefer?

Reply:
2 is fine